lib/moralis.ts

```diff
@@ -40,3 +49,82 @@ export const fetchTransactions = async (
     cursor: transfers.data.cursor,
   };
 };
+
+// fetch very first Glo transaction
+export const fetchFirstGloTransaction = async (
+  address: string,
+  chain: number
+): Promise<Erc20Transaction | null> => {
+  // find first minting transaction
+  const mintingTxs = await findMintingTxs(chain, address);
+
+  // find all incoming transactions
+  let cursor = "";
+  let incomingGloTxs: Erc20Transaction[] = [];
+
+  do {
+    const response = await Moralis.EvmApi.token.getWalletTokenTransfers({
+      address,
+      chain,
+      fromBlock: getFirstGloBlock(chain),
+      cursor: cursor,
+      limit: 100,
+    });
+
+    const incomingGloTxs_ = response.result.filter((tx: Erc20Transaction) => {
+      return (
+        tx.toAddress.equals(address.toLowerCase()) &&
+        tx.contractAddress.equals(chainConfig[chain].toLowerCase())
+      );
+    });
+
+    incomingGloTxs = incomingGloTxs.concat(incomingGloTxs_);
+
+    cursor = response.pagination.cursor as string;
+  } while (cursor !== "" && cursor != null);
```

Comment:
Just set cursor to null at the start so there's one less to compare?

Reply:
Also you could keep looking for GLO tx in the loop, instead of fetching all transactions and then looking for it.